jetty-websocket/websocket-jetty-common/src/main/java/org/eclipse/jetty/websocket/common/JettyWebSocketFrameHandler.java

```diff
@@ -206,6 +206,16 @@ public void onFrame(Frame frame, Callback callback)
             }
         }
 
+        // If we have received a close frame, set state to closed to disallow suspends and resumes.
+        byte opCode = frame.getOpCode();
+        if (opCode == OpCode.CLOSE)
+        {
+            synchronized (this)
```

Comment:
We have just synchronized above, so can this check and change be done in that block?